command/scaling_policy_info.go

```diff
@@ -70,37 +88,75 @@ func (s *ScalingPolicyInfoCommand) Run(args []string) int {
 		return 1
 	}
 
-	if args = flags.Args(); len(args) != 1 {
-		s.Ui.Error("This command takes one argument: <policy_id>")
-		s.Ui.Error(commandErrorText(s))
+	// Get the HTTP client.
+	client, err := s.Meta.Client()
+	if err != nil {
+		s.Ui.Error(fmt.Sprintf("Error initializing client: %s", err))
 		return 1
 	}
 
-	// Get the policy ID.
+	args = flags.Args()
+
+	// Formatted list mode if no policy ID
+	if len(args) == 0 && (json || len(tmpl) > 0) {
+		policies, _, err := client.Scaling().ListPolicies(nil)
+		if err != nil {
+			s.Ui.Error(fmt.Sprintf("Error listing scaling policies: %v", err))
+			return 1
+		}
+		out, err := Format(json, tmpl, policies)
+		if err != nil {
+			s.Ui.Error(err.Error())
+			return 1
+		}
+		s.Ui.Output(out)
+		return 0
+	}
+
+	if len(args) != 1 {
+		s.Ui.Error("This command takes one of the following argument conditions:")
+		s.Ui.Error(" * A single <policy_id>")
+		s.Ui.Error(" * No arguments, with output format specified")
+		s.Ui.Error(commandErrorText(s))
+		return 1
+	}
 	policyID := args[0]
+	if len(policyID) == 1 {
+		s.Ui.Error("Identifier must contain at least two characters.")
+		return 1
+	}
 
-	// Get the HTTP client.
-	client, err := s.Meta.Client()
+	policyID = sanitizeUUIDPrefix(policyID)
+	policies, _, err := client.Scaling().ListPolicies(&api.QueryOptions{
+		Prefix: policyID,
+	})
 	if err != nil {
-		s.Ui.Error(fmt.Sprintf("Error initializing client: %s", err))
+		s.Ui.Error(fmt.Sprintf("Error querying scaling policy: %v", err))
 		return 1
 	}
+	if len(policies) == 0 {
+		s.Ui.Error(fmt.Sprintf("No scaling policies with prefix or id %q found", policyID))
+		return 1
+	}
+	if len(policies) > 1 {
+		out := formatScalingPolicies(policies)
+		s.Ui.Output(fmt.Sprintf("Prefix matched multiple scaling policies\n\n%s", out))
```

Comment:
There are some inconsistencies, but using `Ui.Error` seems to be the most common for this message: https://github.com/hashicorp/nomad/search?p=1&q=%22Prefix+matched+multiple%22

It affects output color:
<img width="859" alt="image" src="https://user-images.githubusercontent.com/775380/106832010-eaf7cf00-665e-11eb-9cb0-0e01f4261fe5.png">

Reply:
cool, thanks 👍